cmd/geth/js.go

```diff
@@ -425,12 +437,13 @@ func (self *jsre) interactive() {
 	}
 }
 
-func hidepassword(input string) string {
-	if passwordRegexp.MatchString(input) {
-		return passwordRepl
-	} else {
-		return input
+func postprocess(input string) (string, bool) {
```

Comment:
how about forgetting this automation entirely, pass interactivity as a parameter to modules and simply enforce interactive use on the console(s).